drivers/SmartThings/zigbee-lock/src/init.lua

```diff
@@ -300,6 +303,88 @@ local function init(driver, device)
   lock_utils.populate_state_from_data(device)
 end
 
+-- The following two functions are from the lock defaults
+-- They are in the base driver temporarily because of BUG2-659
+local lock_state_handler = function(driver, device, value, zb_rx)
+  local attr = capabilities.lock.lock
+  local LOCK_STATE = {
+    [value.NOT_FULLY_LOCKED]     = attr.unknown(),
+    [value.LOCKED]               = attr.locked(),
+    [value.UNLOCKED]             = attr.unlocked()
+  }
+
+  -- this is where we decide whether or not we need to delay our lock event because we've
+  -- observed it coming before the event (or we're starting to compute the timer)
+  local delay = device:get_field(DELAY_LOCK_EVENT) or 100
+  if (delay < MAX_DELAY) then
+    device.thread:call_with_delay(delay+.5, function ()
+      device:emit_event_for_endpoint(zb_rx.address_header.src_endpoint.value, LOCK_STATE[value.value])
+    end)
+  else
+    local socket = require "socket"
+    device:set_field(DELAY_LOCK_EVENT, socket.gettime())
```

Comment:
Is the goal just to set `DELAY_LOCK_EVENT` to something bigger than `MAX_DELAY` or does it really need to be the time? Wondering if we could just set it to `MAX_DELAY` and avoid having to require `socket`